colossus/src/main/scala/colossus/streaming/Pipe.scala

```diff
@@ -131,6 +147,21 @@ class BufferedPipe[T](size: Int) extends Pipe[T, T] {
     }
   }
 
+  /**
+   * Iterate through all buffered items, removing any where the provided
+   * function returns true
+   */
+  def filterScan(f: T => Boolean) {
+    var i = 0
```

Comment:
I think you should be able to use an iterator instead of tracking the index yourself.